packages/web/components/table/cells/pool-composition.tsx

```diff
@@ -1,54 +1,66 @@
+import classNames from "classnames";
 import Image from "next/image";
 import React, { FunctionComponent } from "react";
-import classNames from "classnames";
 import { BaseCell } from "..";
+import { useStore } from "../../../stores";
+import { PoolAssetsIcon } from "../../assets";
 
-export interface PoolCompositionCell {
-  poolId: number;
-  tokenDenoms: string[];
+export interface PoolCompositionCell extends BaseCell {
+  poolId: string;
 }
 
 /** Displays pool composition as a cell in a table.
  *
  *  Accepts the base hover flag.
  */
 export const PoolCompositionCell: FunctionComponent<
-  Partial<BaseCell & PoolCompositionCell>
-> = ({ value, rowHovered, poolId, tokenDenoms }) => (
-  <React.Fragment>
-    {poolId && tokenDenoms ? (
-      <div className="flex gap-1">
-        <div className="flex flex-col items-start text-white-full">
-          <span
-            className={classNames({
-              "text-secondary-200": rowHovered,
-            })}
-          >
-            {tokenDenoms.reduce(
-              (str, denom, index) =>
-                index === tokenDenoms.length - 1
-                  ? str + denom
-                  : str + denom + "/",
-              ""
-            )}
-          </span>
-          <span
-            className={classNames("text-sm font-caption opacity-60", {
-              "text-secondary-600": rowHovered,
-            })}
-          >
-            Pool #{poolId}
-          </span>
+  Partial<PoolCompositionCell>
+> = ({ rowHovered, poolId }) => {
+  const { queriesOsmosisStore, chainStore } = useStore();
+
+  const queryOsmosis = queriesOsmosisStore.get(chainStore.osmosis.chainId);
+
+  if (!poolId) {
+    return <span>No pool</span>;
+  }
+
+  const pool = queryOsmosis.queryGammPools.getPool(poolId);
+
+  return (
+    <React.Fragment>
+      {pool ? (
+        <div className="flex items-center">
+          <PoolAssetsIcon assets={pool.poolAssets} size="sm" />
+          <div className="ml-4 mr-1 flex flex-col items-start text-white-full">
+            <span
+              className={classNames({
+                "text-secondary-200": rowHovered,
+              })}
+            >
+              {pool.poolAssets.length >= 3
+                ? `${pool.poolAssets.length} Token Pool`
+                : pool.poolAssets
+                    .map((asset) => asset.amount.currency.coinDenom)
```

Comment:
For some reason the denom has suddenly been displaying with the channel info.... didn't take the time to dive into the reason why.. ideas?

<img width="563" alt="Screen Shot 2022-03-10 at 11 40 33 AM" src="https://user-images.githubusercontent.com/4606373/157722875-2518da7d-277a-463c-883f-4ff84aa81e72.png">
<img width="1263" alt="Screen Shot 2022-03-10 at 11 41 05 AM" src="https://user-images.githubusercontent.com/4606373/157722940-1e8cdbcf-c271-403c-9695-8959905e74e9.png">

Reply:
initializing `ibcCurrencyRegistrar` is broken after `@keplr-wallet/stores` updated to 0.9.12.
It needs to contain cosmwasm query. I'm gonna fix it.